src/MongoDB.Driver/FilterDefinitionBuilder.cs

```diff
@@ -423,7 +423,7 @@ public FilterDefinition<TDocument> ElemMatch<TItem>(Expression<Func<TDocument, I
         /// <returns>An equality filter.</returns>
         public FilterDefinition<TDocument> Eq<TField>(FieldDefinition<TDocument, TField> field, TField value)
         {
-            return new SimpleFilterDefinition<TDocument, TField>(field, value);
+            return new SimpleFilterDefinition<TDocument, TField>(field, value, allowScalarValueForArrayField: true);
```

Comment:
For all of the filter operators where a scalar value is acceptable for an array field we set allowScalarValueForArrayField to true.

For all others it defaults to false.